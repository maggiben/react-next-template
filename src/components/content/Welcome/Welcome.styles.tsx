import styled from 'styled-components';

export const Container = styled.div`
  position: relative;
`;

export const Background = styled.div`
  height: 70vh;
`;

export const Overlay = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  justify-content: center;
  align-items: center;
`;
export const ContentWrapper = styled.div`

`;
