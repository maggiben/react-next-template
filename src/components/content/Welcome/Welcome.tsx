import styled from "styled-components";
import { useTranslation } from 'react-i18next';
import { Heading, TextBody } from '@fravega-it/bumeran-ds-fvg'

const Block = styled.div`
  display: block;
  width: 100%;
`;

const Welcome = () => {
  const { t } = useTranslation();
  return (
    <Block>
      <Heading size="xl">{t('welcome message')}</Heading>
      <TextBody color="neutral" as="span">{t('tool usage')}</TextBody>
    </Block>
  );
};

  export default Welcome;
