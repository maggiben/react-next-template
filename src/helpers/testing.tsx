import { default as baseRenderer } from 'react-test-renderer';
import { render, RenderResult } from '@testing-library/react';
import i18n from '@utils/i18n';
import { I18nextProvider } from 'react-i18next';
import { RecoilRoot } from 'recoil';
import { ThemeProvider } from 'styled-components';
import { theme, IDefaultTheme } from '@fravega-it/bumeran-ds-fvg';
import React, { ReactElement } from 'react';
import '@testing-library/jest-dom'

/**
 * Helper used to render components for testing.
 * The function appends the theme to the children.
 * */
interface IRenderer {
  create: (children: ReactElement) => RenderResult;
  baseRenderer: (children: ReactElement) => baseRenderer.ReactTestRenderer;
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme) => RenderResult;
  renderWithI18n: (children: ReactElement) => RenderResult;
}

export const renderer: IRenderer = {
  create: (children: ReactElement): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  baseRenderer: (children: ReactElement): baseRenderer.ReactTestRenderer =>
    baseRenderer.create(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithI18n: (children: ReactElement): RenderResult => 
    render(
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
      </ThemeProvider>
  ),
};