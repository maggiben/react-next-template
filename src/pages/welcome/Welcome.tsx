import { useEffect } from 'react';
import Content from '@components/content';
import ClientForm from '@components/forms'
import Head from 'next/head';
import React from 'react';
import { useRouter } from 'next/router';

export default function Welcome(): JSX.Element {
    const router = useRouter();
    // This function will be called whenever the query parameters change
    useEffect(() => {
      // Access the query parameters from the router object
      const { cuid, documentType, documentNumber } = router.query;

      // Perform any actions based on the query parameters
      // eslint-disable-next-line no-console
      console.log('Query parameters changed:', cuid, documentType, documentNumber);
    }, [router.query]);
        return (
        <>
            <Head>
                <meta name="description" content="Frávega template app" />
                <meta
                    name="viewport"
                    content="width=device-width, initial-scale=1"
                />
                <title>Frávega Template App</title>
            </Head>

            <ClientForm />
            <Content />
        </>
    );
}
