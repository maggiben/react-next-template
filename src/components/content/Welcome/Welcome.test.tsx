import React from 'react';
import { renderer } from '@helpers/testing';
import '@testing-library/jest-dom';
import Welcome from './Welcome';
import { expect } from '@jest/globals';

describe('Welcome', () => {
  it('renders the Welcome component', () => {
    const tree = renderer.create(<Welcome />);
    expect(tree).toMatchSnapshot();
  });
});
