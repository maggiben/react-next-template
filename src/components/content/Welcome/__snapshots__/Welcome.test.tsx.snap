// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders the Welcome component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Welcome__Block-sc-wpgyj9-0 bKeliw"
      >
        <div
          aria-level="2"
          class="sc-kAyceB ceJMiP"
          role="heading"
        >
          Te damos la bienvenida al buscador de clientes
        </div>
        <span
          class="sc-iGgWBj emVxnH"
          color="neutral"
        >
          Podés encontrar clientes buscando por tipo y número de documento o Email
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Welcome__Block-sc-wpgyj9-0 bKeliw"
    >
      <div
        aria-level="2"
        class="sc-kAyceB ceJMiP"
        role="heading"
      >
        Te damos la bienvenida al buscador de clientes
      </div>
      <span
        class="sc-iGgWBj emVxnH"
        color="neutral"
      >
        Podés encontrar clientes buscando por tipo y número de documento o Email
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
